namui-cli/src/services/rust_build_service.rs

```diff
@@ -124,12 +124,18 @@ impl CancelableBuilder {
                     match spawned_process.try_wait()? {
                         None => {}
                         Some(exit_status) => {
-                            match parse_cargo_build_result(
-                                stdout_reading_thread
-                                    .join()
-                                    .expect("fail to get stdout from thread")
-                                    .as_bytes(),
-                            ) {
+                            let cargo_outputs = stdout_reading_thread
+                                .join()
+                                .expect("fail to get stdout from thread");
+
+                            if cargo_outputs.is_empty() {
+                                return Err(format!(
+                                    "cargo build failed {stderr}",
+                                    stderr = stderr_reading_thread.join().unwrap()
+                                )
+                                .into());
+                            }
```

Comment:
I just found a new bug about this code. If it failed to run cargo, cargo output's length become 0, but previous code didn't check and just return Ok.